general/front-end-configs.md

```diff
@@ -0,0 +1,183 @@
+# Avenga Frontend Configs
+
+This file contains baseline configurations for Avenga tools commonly used when developing frontends.
+
+Table of contents:
+
+- [Editorconfig](#editorconfig)
+- [Prettier](#prettier)
+- [Stylelint](#stylelint)
+- [ESLint](#eslint)
+- [Setting up Git-Hooks](#git-hooks)
+
+## Editorconfig
+
+All popular editors support [EditorConfig](https://editorconfig.org/) as a way to share common settings like encoding, indentation style or which character to use for line-endings. Throughout our projects we use this config:
+
+```ini
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+insert_final_newline = true
+trim_trailing_whitespace = true
+```
```

Comment:
```
[*.md]
max_line_length = 120
insert_final_newline = false
trim_trailing_whitespace = false
```

aside from the max line length which is highly debatable
md files don't like trimming of final newline because it breaks formating
and a final newline is also not necessarily what we want in a markdown file